cmd/kube-controller-manager/app/core.go

```diff
@@ -265,6 +267,8 @@ func startPersistentVolumeBinderController(ctx ControllerContext) (http.Handler,
 		PodInformer:               ctx.InformerFactory.Core().V1().Pods(),
 		NodeInformer:              ctx.InformerFactory.Core().V1().Nodes(),
 		EnableDynamicProvisioning: ctx.ComponentConfig.PersistentVolumeBinderController.VolumeConfiguration.EnableDynamicProvisioning,
+		CSIMigratedPluginManager:  volume.NewCSIMigratedPluginManager(csiTranslator),
+		CSITranslator:             csiTranslator,
```

Comment:
I can move down the CSIMigratedPluginManager and CSITranslator creation down to the level of the controller initializers rather than creating them at the top (in Kubernetes Controller Manager and Kubelet and passing them down).

We decided against merging the two into VolumePluginManager [earlier](https://github.com/kubernetes/kubernetes/pull/83098#discussion_r331283326)

Reply:
> I can move down the CSIMigratedPluginManager and CSITranslator creation down to the level of the controller initializers rather than creating them at the top (in Kubernetes Controller Manager and Kubelet and passing them down).

Please do.

> We decided against merging the CSIMigratedPluginManager into VolumePluginManager earlier due to difference in functionality.

Ack

Reply:
Done. I have left the Expand Controller init at the top level. For AD and PV controller, moved down the CSI Migration Manager and Translator init one level down below the KCM and Kubelet.